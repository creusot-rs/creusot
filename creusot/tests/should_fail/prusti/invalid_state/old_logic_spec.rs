extern crate creusot_contracts;
use creusot_contracts::prusti_prelude::*;

#[logic]
#[ensures(result == old(x))]
fn test(x: u32) -> u32 {
    x
}
