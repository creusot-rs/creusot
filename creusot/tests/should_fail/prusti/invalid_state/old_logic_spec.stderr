error[creusot]: 'pre can only be used in specifications of program functions
 --> old_logic_spec.rs:5:21
  |
5 | #[ensures(result == old(x))]
  |                     ^^^^^^

error: aborting due to previous error

